In [1]:
import nltk

In [2]:
import string

In [3]:
from collections import Counter

In [4]:
f = open('Data/a1.txt', 'r')

def get_tokens():
    text = f.read()
    lowers = text.lower()
    no_punctuation = text.translate(str.maketrans('','',string.punctuation))
    tokens = nltk.word_tokenize(no_punctuation)
    return tokens

tokens1 = get_tokens()
count1 = Counter(tokens1)

In [5]:
count1.most_common(50)

[('the', 83),
 ('and', 65),
 ('Gyarados', 62),
 ('Pokémon', 53),
 ('in', 49),
 ('of', 49),
 ('a', 41),
 ('it', 35),
 ('to', 32),
 ('is', 27),
 ('as', 25),
 ('by', 14),
 ('one', 14),
 ('Magikarp', 13),
 ('has', 13),
 ('In', 13),
 ('Red', 12),
 ('its', 11),
 ('but', 11),
 ('was', 11),
 ('for', 10),
 ('Blue', 9),
 ('that', 8),
 ('into', 8),
 ('on', 8),
 ('series', 7),
 ('It', 7),
 ('most', 7),
 ('first', 6),
 ('appeared', 6),
 ('been', 6),
 ('Misty', 6),
 ('described', 6),
 ('their', 6),
 ('used', 6),
 ('at', 6),
 ('Game', 5),
 ('games', 5),
 ('later', 5),
 ('evolved', 5),
 ('until', 5),
 ('appears', 5),
 ('red', 5),
 ('Silver', 5),
 ('called', 5),
 ('game', 5),
 ('This', 5),
 ('from', 5),
 ('battle', 5),
 ('player', 5)]

In [6]:
len(tokens1)

1672

In [7]:
len(count1)

662

In [8]:
len(set(tokens1))

662

1672 tokens in a1.txt

662 unique tokens in a1.txt

In [9]:
f = open('Data/a2.txt', 'r')

tokens2 = get_tokens()
count2 = Counter(tokens2)

In [10]:
count2.most_common(50)

[('the', 285),
 ('and', 134),
 ('in', 110),
 ('of', 110),
 ('Ash', 106),
 ('to', 100),
 ('a', 97),
 ('Pokémon', 88),
 ('his', 64),
 ('as', 54),
 ('is', 43),
 ('with', 40),
 ('he', 33),
 ('was', 33),
 ('anime', 27),
 ('from', 27),
 ('The', 27),
 ('by', 25),
 ('League', 25),
 ('character', 24),
 ('that', 24),
 ('series', 23),
 ('on', 23),
 ('In', 23),
 ('has', 21),
 ('Pikachu', 21),
 ('all', 20),
 ('also', 20),
 ('him', 20),
 ('for', 18),
 ('which', 16),
 ('He', 15),
 ('Ashs', 15),
 ('Team', 15),
 ('first', 14),
 ('more', 14),
 ('had', 14),
 ('trainer', 14),
 ('their', 14),
 ('episode', 13),
 ('characters', 12),
 ('one', 12),
 ('new', 12),
 ('Ketchum', 11),
 ('show', 11),
 ('an', 11),
 ('rival', 11),
 ('they', 11),
 ('it', 11),
 ('against', 11)]

In [11]:
len(tokens2)

4223

In [12]:
len(count2)

1338

In [13]:
len(set(tokens2))

1338

4223 tokens in a2.txt

1338 unique tokens in a2.txt

In [14]:
f = open('Data/a3.txt', 'r')

tokens3 = get_tokens()
count3 = Counter(tokens3)

In [15]:
count3.most_common(50)

[('the', 96),
 ('of', 65),
 ('a', 59),
 ('and', 59),
 ('Jigglypuff', 55),
 ('to', 52),
 ('Pokémon', 49),
 ('in', 42),
 ('it', 32),
 ('as', 28),
 ('is', 28),
 ('by', 25),
 ('its', 24),
 ('was', 15),
 ('series', 14),
 ('on', 14),
 ('In', 13),
 ('has', 13),
 ('which', 13),
 ('The', 12),
 ('for', 12),
 ('with', 11),
 ('when', 10),
 ('anime', 10),
 ('one', 10),
 ('that', 10),
 ('Jigglypuffs', 10),
 ('games', 9),
 ('character', 9),
 ('her', 9),
 ('their', 8),
 ('first', 7),
 ('into', 7),
 ('or', 6),
 ('evolves', 6),
 ('several', 6),
 ('It', 6),
 ('are', 6),
 ('sleep', 6),
 ('who', 6),
 ('asleep', 6),
 ('song', 6),
 ('popular', 6),
 ('species', 5),
 ('Game', 5),
 ('franchise', 5),
 ('appeared', 5),
 ('Wigglytuff', 5),
 ('been', 5),
 ('characters', 5)]

In [16]:
len(tokens3)

1905

In [17]:
len(count3)

744

In [18]:
len(set(tokens3))

744

1905 tokens in a3.txt

744 unique tokens in a3.txt

In [19]:
f = open('Data/a4.txt', 'r')

tokens4 = get_tokens()
count4 = Counter(tokens4)

In [20]:
count4.most_common(50)

[('the', 174),
 ('and', 112),
 ('Charizard', 101),
 ('Pokémon', 81),
 ('of', 75),
 ('a', 61),
 ('to', 58),
 ('in', 57),
 ('as', 37),
 ('is', 36),
 ('with', 26),
 ('it', 25),
 ('Ash', 24),
 ('its', 22),
 ('by', 20),
 ('Mega', 20),
 ('one', 20),
 ('In', 18),
 ('that', 16),
 ('Ashs', 16),
 ('from', 15),
 ('character', 15),
 ('was', 15),
 ('his', 15),
 ('Red', 14),
 ('Charmander', 14),
 ('an', 14),
 ('has', 13),
 ('battle', 13),
 ('for', 12),
 ('who', 11),
 ('Charmeleon', 11),
 ('X', 11),
 ('which', 11),
 ('most', 11),
 ('their', 11),
 ('into', 11),
 ('Blue', 10),
 ('Charizards', 10),
 ('on', 10),
 ('can', 10),
 ('It', 9),
 ('out', 9),
 ('when', 9),
 ('The', 9),
 ('he', 9),
 ('against', 9),
 ('Y', 8),
 ('series', 8),
 ('appears', 8)]

In [21]:
len(tokens4)

2885

In [22]:
len(count4)

1033

In [23]:
len(set(tokens4))

1033

2885 tokens in a4.txt

1033 unique tokens in a4.txt

In [24]:
f = open('Data/a5.txt', 'r')

tokens5 = get_tokens()
count5 = Counter(tokens5)

In [25]:
count5.most_common(50)

[('the', 174),
 ('Mewtwo', 108),
 ('and', 103),
 ('of', 80),
 ('in', 73),
 ('a', 69),
 ('to', 67),
 ('Pokémon', 59),
 ('as', 58),
 ('it', 39),
 ('by', 36),
 ('is', 35),
 ('its', 34),
 ('In', 27),
 ('for', 24),
 ('character', 22),
 ('was', 21),
 ('The', 21),
 ('which', 19),
 ('from', 17),
 ('games', 17),
 ('on', 17),
 ('that', 17),
 ('with', 16),
 ('Mewtwos', 13),
 ('film', 13),
 ('Mew', 12),
 ('Japanese', 11),
 ('anime', 11),
 ('been', 11),
 ('has', 11),
 ('Mega', 11),
 ('his', 11),
 ('voiced', 10),
 ('have', 10),
 ('Red', 9),
 ('other', 9),
 ('such', 9),
 ('an', 9),
 ('be', 9),
 ('also', 9),
 ('one', 8),
 ('series', 8),
 ('characters', 8),
 ('appears', 8),
 ('after', 8),
 ('Smash', 8),
 ('Nintendo', 7),
 ('player', 7),
 ('being', 7)]

In [26]:
len(tokens5)

2998

In [27]:
len(count5)

1113

In [28]:
len(set(tokens5))

1113

2998 tokens in a5.txt

1113 unique tokens in a5.txt

In [29]:
f = open('Data/a6.txt', 'r')

tokens6 = get_tokens()
count6 = Counter(tokens6)

In [30]:
count6.most_common(50)

[('the', 75),
 ('Meowth', 55),
 ('and', 50),
 ('in', 45),
 ('Pokémon', 44),
 ('a', 40),
 ('to', 38),
 ('as', 31),
 ('of', 29),
 ('that', 18),
 ('is', 16),
 ('it', 16),
 ('anime', 12),
 ('was', 11),
 ('its', 11),
 ('which', 10),
 ('In', 10),
 ('not', 10),
 ('by', 9),
 ('one', 9),
 ('has', 9),
 ('him', 9),
 ('their', 9),
 ('have', 8),
 ('on', 8),
 ('he', 8),
 ('his', 8),
 ('from', 8),
 ('first', 7),
 ('games', 7),
 ('an', 7),
 ('at', 7),
 ('character', 7),
 ('for', 7),
 ('species', 6),
 ('Game', 6),
 ('series', 6),
 ('The', 6),
 ('Japanese', 5),
 ('language', 5),
 ('Nintendo', 5),
 ('most', 5),
 ('into', 5),
 ('Team', 5),
 ('Rocket', 5),
 ('speak', 5),
 ('characters', 5),
 ('would', 5),
 ('appearance', 5),
 ('appears', 5)]

In [31]:
len(tokens6)

1542

In [32]:
len(count6)

639

In [33]:
len(set(count6))

639

1542 tokens in a6.txt

639 unique tokens in a6.txt

In [34]:
f = open('Data/a7.txt', 'r')

tokens7 = get_tokens()
count7 = Counter(tokens7)

In [35]:
count7.most_common(50)

[('the', 76),
 ('and', 55),
 ('in', 48),
 ('Gengar', 44),
 ('Pokémon', 41),
 ('of', 39),
 ('a', 33),
 ('to', 32),
 ('its', 23),
 ('it', 20),
 ('as', 17),
 ('is', 15),
 ('by', 15),
 ('with', 13),
 ('was', 13),
 ('that', 12),
 ('In', 11),
 ('for', 11),
 ('Blue', 9),
 ('Red', 8),
 ('best', 7),
 ('an', 7),
 ('The', 7),
 ('Nintendo', 6),
 ('Mega', 6),
 ('also', 6),
 ('from', 6),
 ('Haunter', 6),
 ('species', 5),
 ('Game', 5),
 ('It', 5),
 ('appeared', 5),
 ('games', 5),
 ('Japanese', 5),
 ('have', 5),
 ('Ghost', 5),
 ('which', 5),
 ('game', 5),
 ('Gym', 5),
 ('her', 5),
 ('A', 5),
 ('their', 4),
 ('name', 4),
 ('later', 4),
 ('are', 4),
 ('has', 4),
 ('noted', 4),
 ('children', 4),
 ('appearance', 4),
 ('IGN', 4)]

In [36]:
len(tokens7)

1421

In [37]:
len(count7)

623

In [38]:
len(set(tokens7))

623

1421 tokens in a7.txt

623 unique tokens in a7.txt

In [39]:
f = open('Data/a8.txt', 'r')

tokens8 = get_tokens()
count8 = Counter(tokens8)

In [40]:
count8.most_common(50)

[('the', 95),
 ('and', 58),
 ('in', 51),
 ('Jynx', 49),
 ('of', 44),
 ('Pokémon', 36),
 ('a', 34),
 ('to', 31),
 ('as', 26),
 ('design', 15),
 ('with', 15),
 ('it', 15),
 ('is', 12),
 ('by', 12),
 ('on', 12),
 ('was', 11),
 ('an', 11),
 ('games', 10),
 ('episodes', 10),
 ('their', 9),
 ('In', 9),
 ('Jynxs', 8),
 ('described', 8),
 ('its', 8),
 ('from', 8),
 ('which', 8),
 ('that', 8),
 ('her', 8),
 ('later', 7),
 ('appearance', 7),
 ('purple', 7),
 ('have', 7),
 ('Game', 6),
 ('The', 6),
 ('were', 6),
 ('episode', 6),
 ('first', 5),
 ('appeared', 5),
 ('Red', 5),
 ('Japanese', 5),
 ('after', 5),
 ('anime', 5),
 ('black', 5),
 ('one', 5),
 ('for', 5),
 ('can', 5),
 ('series', 5),
 ('controversy', 5),
 ('also', 5),
 ('Nintendo', 4)]

In [41]:
len(tokens8)

1591

In [42]:
len(count8)

698

In [43]:
len(set(tokens8))

698

1591 tokens in a8.txt

698 unique tokens in a8.txt

In [44]:
f = open('Data/a9.txt', 'r')

tokens9 = get_tokens()
count9 = Counter(tokens9)

In [45]:
count9.most_common(50)

[('a', 59),
 ('Togepi', 56),
 ('the', 53),
 ('and', 50),
 ('to', 39),
 ('Pokémon', 37),
 ('of', 32),
 ('in', 31),
 ('it', 25),
 ('is', 23),
 ('as', 21),
 ('its', 20),
 ('that', 14),
 ('on', 13),
 ('In', 12),
 ('which', 10),
 ('has', 10),
 ('first', 9),
 ('Gold', 9),
 ('with', 9),
 ('Togetic', 9),
 ('Togekiss', 9),
 ('egg', 8),
 ('was', 8),
 ('Misty', 7),
 ('anime', 6),
 ('also', 6),
 ('who', 6),
 ('into', 6),
 ('appeared', 5),
 ('It', 5),
 ('Silver', 5),
 ('red', 5),
 ('blue', 5),
 ('one', 5),
 ('an', 5),
 ('from', 5),
 ('for', 4),
 ('games', 4),
 ('body', 4),
 ('evolution', 4),
 ('after', 4),
 ('appearance', 4),
 ('called', 4),
 ('player', 4),
 ('design', 4),
 ('Togepis', 4),
 ('will', 4),
 ('back', 4),
 ('her', 4)]

In [46]:
len(tokens9)

1401

In [47]:
len(count9)

602

In [48]:
len(set(tokens9))

602

1401 tokens in a9.txt

602 unique tokens in a9.txt

In [49]:
f = open('Data/a10.txt', 'r')

tokens10 = get_tokens()
count10 = Counter(tokens10)

In [50]:
count10.most_common(50)

[('the', 65),
 ('and', 52),
 ('Snorlax', 42),
 ('a', 32),
 ('of', 31),
 ('Pokémon', 30),
 ('it', 28),
 ('to', 27),
 ('in', 25),
 ('is', 19),
 ('as', 14),
 ('by', 14),
 ('In', 14),
 ('was', 11),
 ('its', 11),
 ('has', 10),
 ('Red', 8),
 ('most', 8),
 ('for', 8),
 ('games', 7),
 ('character', 7),
 ('one', 7),
 ('that', 7),
 ('called', 7),
 ('when', 6),
 ('also', 6),
 ('but', 6),
 ('with', 6),
 ('later', 5),
 ('been', 5),
 ('Ash', 5),
 ('IGN', 5),
 ('an', 5),
 ('known', 4),
 ('first', 4),
 ('It', 4),
 ('until', 4),
 ('powerful', 4),
 ('which', 4),
 ('their', 4),
 ('children', 4),
 ('sleeping', 4),
 ('from', 4),
 ('time', 4),
 ('The', 4),
 ('player', 4),
 ('must', 4),
 ('up', 4),
 ('used', 4),
 ('on', 4)]

In [51]:
len(tokens10)

1247

In [52]:
len(count10)

560

In [53]:
len(set(tokens10))

560

1247 tokens in a10.txt

560 unique tokens in a10.txt

In [54]:
allTokens = len(tokens1) + len(tokens2) + len(tokens3) + len(tokens4) + len(tokens5) + len(tokens6) + len(tokens7) + len(tokens8) + len(tokens9) + len(tokens10)

In [55]:
print(allTokens)

20885


In [56]:
allUniqueTokens = len(count1) + len(count2) + len(count3) + len(count4) + len(count5) + len(count6) + len(count7) + len(count8) + len(count9) + len(count10)

In [57]:
# this is not unique tokens only - some are unique in their documents but repeat in others
print(allUniqueTokens)

8012


In [58]:
allData = ['Data/a1.txt', 'Data/a2.txt', 'Data/a3.txt', 'Data/a4.txt', 'Data/a5.txt', 'Data/a6.txt', 'Data/a7.txt', 'Data/a8.txt', 'Data/a9.txt', 'Data/a10.txt']

with open('allData.txt', 'w') as outfile:
    for names in allData:
        with open(names) as infile:
            outfile.write(infile.read())
        outfile.write("\n")

In [59]:
f = open('allData.txt', 'r')

tokensAll = get_tokens()
countAll = Counter(tokensAll)

In [60]:
len(tokensAll)

20885

In [61]:
len(countAll)

4144

In [62]:
len(set(tokensAll))

4144

20885 tokens in all 10 documents

4144 unique tokens in all 10 documents

In [63]:
from nltk.corpus import stopwords

In [64]:
filtered1 = [w for w in tokens1 if not w in stopwords.words('english')]
fcount1 = Counter(filtered1)

In [65]:
fcount1.most_common(50)

[('Gyarados', 62),
 ('Pokémon', 53),
 ('one', 14),
 ('Magikarp', 13),
 ('In', 13),
 ('Red', 12),
 ('Blue', 9),
 ('series', 7),
 ('It', 7),
 ('first', 6),
 ('appeared', 6),
 ('Misty', 6),
 ('described', 6),
 ('used', 6),
 ('Game', 5),
 ('games', 5),
 ('later', 5),
 ('evolved', 5),
 ('appears', 5),
 ('red', 5),
 ('Silver', 5),
 ('called', 5),
 ('game', 5),
 ('This', 5),
 ('battle', 5),
 ('player', 5),
 ('Team', 5),
 ('various', 4),
 ('The', 4),
 ('flying', 4),
 ('type', 4),
 ('team', 4),
 ('Chinese', 4),
 ('IGN', 4),
 ('species', 3),
 ('video', 3),
 ('Green', 3),
 ('merchandise', 3),
 ('Japanese', 3),
 ('form', 3),
 ('well', 3),
 ('known', 3),
 ('around', 3),
 ('times', 3),
 ('anime', 3),
 ('trainers', 3),
 ('Lance', 3),
 ('Crasher', 3),
 ('Wake', 3),
 ('generally', 3)]

In [66]:
len(filtered1)

1027

In [67]:
len(fcount1)

588

In [68]:
len(set(filtered1))

588

1027 tokens in a1.txt after removing stopwords

588 unique tokens in a1.txt after removing stopwords

In [69]:
filtered2 = [w for w in tokens2 if not w in stopwords.words('english')]
fcount2 = Counter(filtered2)
fcount2.most_common(50)

[('Ash', 106),
 ('Pokémon', 88),
 ('anime', 27),
 ('The', 27),
 ('League', 25),
 ('character', 24),
 ('series', 23),
 ('In', 23),
 ('Pikachu', 21),
 ('also', 20),
 ('He', 15),
 ('Ashs', 15),
 ('Team', 15),
 ('first', 14),
 ('trainer', 14),
 ('episode', 13),
 ('characters', 12),
 ('one', 12),
 ('new', 12),
 ('Ketchum', 11),
 ('show', 11),
 ('rival', 11),
 ('well', 10),
 ('However', 10),
 ('badges', 10),
 ('8', 10),
 ('Japanese', 9),
 ('Taylor', 9),
 ('Red', 9),
 ('I', 9),
 ('Frontier', 9),
 ('Kalos', 9),
 ('games', 8),
 ('many', 8),
 ('defeated', 8),
 ('Top', 8),
 ('Conference', 8),
 ('name', 7),
 ('game', 7),
 ('named', 7),
 ('Gary', 7),
 ('Brock', 7),
 ('Orange', 7),
 ('Sinnoh', 7),
 ('Greninja', 7),
 ('parody', 7),
 ('Satoshi', 6),
 ('season', 6),
 ('Blue', 6),
 ('time', 6)]

In [70]:
len(filtered2)

2537

In [71]:
len(fcount2)

1236

In [72]:
len(set(filtered2))

1236

2537 tokens in a2.txt after removing stopwords

1236 unique tokens in a2.txt after removing stopwords

In [73]:
filtered3 = [w for w in tokens3 if not w in stopwords.words('english')]
fcount3 = Counter(filtered3)
fcount3.most_common(50)

[('Jigglypuff', 55),
 ('Pokémon', 49),
 ('series', 14),
 ('In', 13),
 ('The', 12),
 ('anime', 10),
 ('one', 10),
 ('Jigglypuffs', 10),
 ('games', 9),
 ('character', 9),
 ('first', 7),
 ('evolves', 6),
 ('several', 6),
 ('It', 6),
 ('sleep', 6),
 ('asleep', 6),
 ('song', 6),
 ('popular', 6),
 ('species', 5),
 ('Game', 5),
 ('franchise', 5),
 ('appeared', 5),
 ('Wigglytuff', 5),
 ('characters', 5),
 ('children', 5),
 ('sing', 5),
 ('female', 5),
 ('Red', 4),
 ('Moon', 4),
 ('appearance', 4),
 ('featured', 4),
 ('plush', 4),
 ('game', 4),
 ('like', 4),
 ('pink', 4),
 ('singing', 4),
 ('two', 4),
 ('also', 4),
 ('Nintendo', 3),
 ('Freaks', 3),
 ('video', 3),
 ('Blue', 3),
 ('printed', 3),
 ('Japanese', 3),
 ('action', 3),
 ('Igglybuff', 3),
 ('reaches', 3),
 ('certain', 3),
 ('point', 3),
 ('happiness', 3)]

In [74]:
len(filtered3)

1151

In [75]:
len(fcount3)

676

In [76]:
len(set(filtered3))

676

1151 tokens in a3.txt after removing stopwords

676 unique tokens in a3.txt after removing stopwords

In [77]:
filtered4 = [w for w in tokens4 if not w in stopwords.words('english')]
fcount4 = Counter(filtered4)
fcount4.most_common(50)

[('Charizard', 101),
 ('Pokémon', 81),
 ('Ash', 24),
 ('Mega', 20),
 ('one', 20),
 ('In', 18),
 ('Ashs', 16),
 ('character', 15),
 ('Red', 14),
 ('Charmander', 14),
 ('battle', 13),
 ('Charmeleon', 11),
 ('X', 11),
 ('Blue', 10),
 ('Charizards', 10),
 ('It', 9),
 ('The', 9),
 ('Y', 8),
 ('series', 8),
 ('appears', 8),
 ('anime', 7),
 ('evolved', 7),
 ('characters', 7),
 ('Team', 7),
 ('Smash', 7),
 ('League', 7),
 ('first', 6),
 ('games', 6),
 ('They', 6),
 ('Evolution', 6),
 ('Super', 6),
 ('game', 6),
 ('stated', 6),
 ('popular', 6),
 ('powerful', 6),
 ('three', 6),
 ('video', 5),
 ('form', 5),
 ('two', 5),
 ('main', 5),
 ('team', 5),
 ('called', 5),
 ('children', 5),
 ('type', 5),
 ('also', 5),
 ('Bros', 5),
 ('Pikachu', 5),
 ('Reds', 5),
 ('Saur', 5),
 ('editor', 5)]

In [78]:
len(filtered4)

1792

In [79]:
len(fcount4)

947

In [80]:
len(set(filtered4))

947

1792 tokens in a4.txt after removing stopwords

947 unique tokens in a4.txt after removing stopwords

In [81]:
filtered5 = [w for w in tokens5 if not w in stopwords.words('english')]
fcount5 = Counter(filtered5)
fcount5.most_common(50)

[('Mewtwo', 108),
 ('Pokémon', 59),
 ('In', 27),
 ('character', 22),
 ('The', 21),
 ('games', 17),
 ('Mewtwos', 13),
 ('film', 13),
 ('Mew', 12),
 ('Japanese', 11),
 ('anime', 11),
 ('Mega', 11),
 ('voiced', 10),
 ('Red', 9),
 ('also', 9),
 ('one', 8),
 ('series', 8),
 ('characters', 8),
 ('appears', 8),
 ('Smash', 8),
 ('Nintendo', 7),
 ('player', 7),
 ('Super', 7),
 ('Bros', 7),
 ('Fuji', 7),
 ('franchise', 6),
 ('first', 6),
 ('video', 6),
 ('Blue', 6),
 ('later', 6),
 ('Strikes', 6),
 ('Mews', 6),
 ('Team', 6),
 ('Giovanni', 6),
 ('English', 5),
 ('Back', 5),
 ('villain', 5),
 ('clone', 5),
 ('appearance', 5),
 ('As', 5),
 ('DNA', 5),
 ('abilities', 5),
 ('Y', 5),
 ('events', 5),
 ('featured', 5),
 ('Rocket', 5),
 ('Ash', 5),
 ('described', 5),
 ('media', 4),
 ('appeared', 4)]

In [82]:
len(filtered5)

1875

In [83]:
len(fcount5)

1031

In [84]:
len(set(filtered5))

1031

1875 tokens in a5.txt after removing stopwords

1031 unique tokens in a5.txt after removing stopwords

In [85]:
filtered6 = [w for w in tokens6 if not w in stopwords.words('english')]
fcount6 = Counter(filtered6)
fcount6.most_common(50)

[('Meowth', 55),
 ('Pokémon', 44),
 ('anime', 12),
 ('In', 10),
 ('one', 9),
 ('first', 7),
 ('games', 7),
 ('character', 7),
 ('species', 6),
 ('Game', 6),
 ('series', 6),
 ('The', 6),
 ('Japanese', 5),
 ('language', 5),
 ('Nintendo', 5),
 ('Team', 5),
 ('Rocket', 5),
 ('speak', 5),
 ('characters', 5),
 ('would', 5),
 ('appearance', 5),
 ('appears', 5),
 ('franchise', 4),
 ('appeared', 4),
 ('Red', 4),
 ('Blue', 4),
 ('episode', 4),
 ('ability', 4),
 ('human', 4),
 ('Meowzie', 4),
 ('two', 4),
 ('describing', 4),
 ('game', 4),
 ('GamesRadar', 4),
 ('described', 4),
 ('coins', 4),
 ('editor', 4),
 ('known', 3),
 ('version', 3),
 ('video', 3),
 ('It', 3),
 ('James', 3),
 ('English', 3),
 ('city', 3),
 ('called', 3),
 ('walk', 3),
 ('legs', 3),
 ('said', 3),
 ('also', 3),
 ('move', 3)]

In [86]:
len(filtered6)

941

In [87]:
len(fcount6)

565

In [88]:
len(set(filtered6))

565

941 tokens in a6.txt after removing stopwords

565 unique tokens in a6.txt after removing stopwords

In [89]:
filtered7 = [w for w in tokens7 if not w in stopwords.words('english')]
fcount7 = Counter(filtered7)
fcount7.most_common(50)

[('Gengar', 44),
 ('Pokémon', 41),
 ('In', 11),
 ('Blue', 9),
 ('Red', 8),
 ('best', 7),
 ('The', 7),
 ('Nintendo', 6),
 ('Mega', 6),
 ('also', 6),
 ('Haunter', 6),
 ('species', 5),
 ('Game', 5),
 ('It', 5),
 ('appeared', 5),
 ('games', 5),
 ('Japanese', 5),
 ('Ghost', 5),
 ('game', 5),
 ('Gym', 5),
 ('A', 5),
 ('name', 4),
 ('later', 4),
 ('noted', 4),
 ('children', 4),
 ('appearance', 4),
 ('IGN', 4),
 ('series', 4),
 ('characters', 4),
 ('first', 4),
 ('form', 4),
 ('shadow', 4),
 ('City', 4),
 ('Ash', 4),
 ('used', 4),
 ('franchise', 3),
 ('popular', 3),
 ('praised', 3),
 ('one', 3),
 ('character', 3),
 ('due', 3),
 ('shadows', 3),
 ('anime', 3),
 ('Gengars', 3),
 ('Agatha', 3),
 ('Leader', 3),
 ('made', 3),
 ('favourite', 3),
 ('design', 3),
 ('named', 3)]

In [90]:
len(filtered7)

886

In [91]:
len(fcount7)

561

In [92]:
len(set(filtered7))

561

886 tokens in a7.txt after removing stopwords

561 unique tokens in a7.txt after removing stopwords

In [93]:
filtered8 = [w for w in tokens8 if not w in stopwords.words('english')]
fcount8 = Counter(filtered8)
fcount8.most_common(50)

[('Jynx', 49),
 ('Pokémon', 36),
 ('design', 15),
 ('games', 10),
 ('episodes', 10),
 ('In', 9),
 ('Jynxs', 8),
 ('described', 8),
 ('later', 7),
 ('appearance', 7),
 ('purple', 7),
 ('Game', 6),
 ('The', 6),
 ('episode', 6),
 ('first', 5),
 ('appeared', 5),
 ('Red', 5),
 ('Japanese', 5),
 ('anime', 5),
 ('black', 5),
 ('one', 5),
 ('series', 5),
 ('controversy', 5),
 ('also', 5),
 ('Nintendo', 4),
 ('various', 4),
 ('English', 4),
 ('blackface', 4),
 ('Smoochum', 4),
 ('people', 4),
 ('Lorelei', 4),
 ('would', 4),
 ('due', 4),
 ('removed', 4),
 ('Rougela', 3),
 ('species', 3),
 ('Sugimori', 3),
 ('Blue', 3),
 ('character', 3),
 ('criticized', 3),
 ('Weatherford', 3),
 ('color', 3),
 ('designs', 3),
 ('racism', 3),
 ('addition', 3),
 ('appears', 3),
 ('Silver', 3),
 ('skin', 3),
 ('form', 3),
 ('However', 3)]

In [94]:
len(filtered8)

989

In [95]:
len(fcount8)

624

In [96]:
len(set(filtered8))

624

989 tokens in a8.txt after removing stopwords

624 unique tokens in a8.txt after removing stopwords

In [97]:
filtered9 = [w for w in tokens9 if not w in stopwords.words('english')]
fcount9 = Counter(filtered9)
fcount9.most_common(50)

[('Togepi', 56),
 ('Pokémon', 37),
 ('In', 12),
 ('first', 9),
 ('Gold', 9),
 ('Togetic', 9),
 ('Togekiss', 9),
 ('egg', 8),
 ('Misty', 7),
 ('anime', 6),
 ('also', 6),
 ('appeared', 5),
 ('It', 5),
 ('Silver', 5),
 ('red', 5),
 ('blue', 5),
 ('one', 5),
 ('games', 4),
 ('body', 4),
 ('evolution', 4),
 ('appearance', 4),
 ('called', 4),
 ('player', 4),
 ('design', 4),
 ('Togepis', 4),
 ('back', 4),
 ('several', 4),
 ('featured', 4),
 ('fans', 4),
 ('character', 3),
 ('shell', 3),
 ('pattern', 3),
 ('stubby', 3),
 ('happiness', 3),
 ('wings', 3),
 ('Diamond', 3),
 ('Pearl', 3),
 ('second', 3),
 ('Shiny', 3),
 ('Stone', 3),
 ('retains', 3),
 ('Its', 3),
 ('requires', 3),
 ('due', 3),
 ('forms', 3),
 ('appears', 3),
 ('since', 3),
 ('Ash', 3),
 ('episode', 3),
 ('Mirage', 3)]

In [98]:
len(filtered9)

867

In [99]:
len(fcount9)

527

In [100]:
len(set(filtered9))

527

867 tokens in a9.txt after removing stopwords

527 unique tokens in a9.txt after removing stopwords

In [101]:
filtered10 = [w for w in tokens10 if not w in stopwords.words('english')]
fcount10 = Counter(filtered10)
fcount10.most_common(50)

[('Snorlax', 42),
 ('Pokémon', 30),
 ('In', 14),
 ('Red', 8),
 ('games', 7),
 ('character', 7),
 ('one', 7),
 ('called', 7),
 ('also', 6),
 ('later', 5),
 ('Ash', 5),
 ('IGN', 5),
 ('known', 4),
 ('first', 4),
 ('It', 4),
 ('powerful', 4),
 ('children', 4),
 ('sleeping', 4),
 ('time', 4),
 ('The', 4),
 ('player', 4),
 ('must', 4),
 ('used', 4),
 ('Pokemon', 4),
 ('species', 3),
 ('video', 3),
 ('Blue', 3),
 ('various', 3),
 ('voiced', 3),
 ('anime', 3),
 ('series', 3),
 ('main', 3),
 ('calling', 3),
 ('Its', 3),
 ('large', 3),
 ('Snorlaxs', 3),
 ('play', 3),
 ('Munchlax', 3),
 ('certain', 3),
 ('roadblock', 3),
 ('Poké', 3),
 ('He', 3),
 ('Lax', 3),
 ('editor', 3),
 ('Japan', 2),
 ('Kabigon', 2),
 ('Nintendo', 2),
 ('Game', 2),
 ('Freaks', 2),
 ('franchise', 2)]

In [102]:
len(filtered10)

761

In [103]:
len(fcount10)

491

In [104]:
len(set(filtered10))

491

761 tokens in a10.txt after removing stopwords

491 unique tokens in a10.txt after removing stopwords

In [105]:
allFiltered = len(filtered1) + len(filtered2) + len(filtered3) + len(filtered4) + len(filtered5) + len(filtered6) + len(filtered7) + len(filtered8) + len(filtered9) + len(filtered10)

In [106]:
print(allFiltered)

12826


In [107]:
allfcount = len(fcount1) + len(fcount2) + len(fcount3) + len(fcount4) + len(fcount5) + len(fcount6) + len(fcount7) + len(fcount8) + len(fcount9) + len(fcount10)

In [108]:
print(allfcount)

7246


In [109]:
filteredAll = [w for w in tokensAll if not w in stopwords.words('english')]
fcountAll = Counter(filteredAll)

In [110]:
len(filteredAll)

12826

In [111]:
len(fcountAll)

4033

In [112]:
len(set(filteredAll))

4033

12826 tokens in all 10 documents after removing stopwords

4033 unique tokens in all 10 documents after removing stopwords

In [113]:
from nltk.stem.porter import *

In [114]:
def stem_tokens(tokens, stemmer):
    stemmed = []
    for item in tokens:
        stemmed.append(stemmer.stem(item))
    return stemmed

stemmer = PorterStemmer()

stemmed1 = stem_tokens(filtered1, stemmer)
scount1 = Counter(stemmed1)
scount1.most_common(50)

[('gyarado', 62),
 ('pokémon', 53),
 ('appear', 20),
 ('red', 18),
 ('game', 15),
 ('one', 15),
 ('magikarp', 13),
 ('In', 13),
 ('evolv', 12),
 ('blue', 12),
 ('team', 9),
 ('seri', 7),
 ('It', 7),
 ('describ', 7),
 ('first', 6),
 ('later', 6),
 ('form', 6),
 ('misti', 6),
 ('trade', 6),
 ('gener', 6),
 ('use', 6),
 ('player', 6),
 ('silver', 5),
 ('fli', 5),
 ('type', 5),
 ('call', 5),
 ('sea', 5),
 ('dragon', 5),
 ('thi', 5),
 ('battl', 5),
 ('ign', 5),
 ('variou', 4),
 ('anim', 4),
 ('known', 4),
 ('time', 4),
 ('trainer', 4),
 ('own', 4),
 ('the', 4),
 ('featur', 4),
 ('charact', 4),
 ('name', 4),
 ('larg', 4),
 ('chines', 4),
 ('evolut', 4),
 ('gain', 4),
 ('speci', 3),
 ('video', 3),
 ('green', 3),
 ('merchandis', 3),
 ('japanes', 3)]

In [115]:
len(stemmed1)

1027

In [116]:
len(scount1)

502

In [117]:
len(set(stemmed1))

502

1027 tokens in a1.txt after removing stopwords and applying stemming

502 unique tokens in a1.txt after removing stopwords and applying stemming

In [118]:
stemmed2 = stem_tokens(filtered2, stemmer)
scount2 = Counter(stemmed2)
scount2.most_common(50)

[('ash', 122),
 ('pokémon', 89),
 ('charact', 37),
 ('anim', 32),
 ('leagu', 28),
 ('the', 27),
 ('seri', 25),
 ('In', 23),
 ('pikachu', 21),
 ('also', 21),
 ('game', 17),
 ('team', 17),
 ('name', 16),
 ('episod', 16),
 ('defeat', 16),
 ('He', 15),
 ('show', 15),
 ('rival', 15),
 ('trainer', 15),
 ('first', 14),
 ('battl', 14),
 ('appear', 13),
 ('red', 12),
 ('one', 12),
 ('travel', 12),
 ('new', 12),
 ('ketchum', 11),
 ('howev', 11),
 ('compet', 11),
 ('badg', 11),
 ('parodi', 11),
 ('well', 10),
 ('season', 10),
 ('make', 10),
 ('8', 10),
 ('confer', 10),
 ('japanes', 9),
 ('voic', 9),
 ('taylor', 9),
 ('I', 9),
 ('power', 9),
 ('frontier', 9),
 ('kalo', 9),
 ('satoshi', 8),
 ('becom', 8),
 ('like', 8),
 ('region', 8),
 ('mani', 8),
 ('top', 8),
 ('greninja', 8)]

In [119]:
len(stemmed2)

2537

In [120]:
len(scount2)

997

In [121]:
len(set(stemmed2))

997

2537 tokens in a2.txt after removing stopwords and applying stemming

997 unique tokens in a2.txt after removing stopwords and applying stemming

In [122]:
stemmed3 = stem_tokens(filtered3, stemmer)
scount3 = Counter(stemmed3)
scount3.most_common(50)

[('jigglypuff', 65),
 ('pokémon', 50),
 ('game', 18),
 ('appear', 17),
 ('charact', 15),
 ('seri', 14),
 ('In', 13),
 ('anim', 12),
 ('the', 12),
 ('sing', 11),
 ('one', 10),
 ('sleep', 9),
 ('song', 8),
 ('first', 7),
 ('evolv', 7),
 ('name', 7),
 ('use', 7),
 ('popular', 7),
 ('sever', 6),
 ('It', 6),
 ('asleep', 6),
 ('speci', 5),
 ('franchis', 5),
 ('video', 5),
 ('blue', 5),
 ('wigglytuff', 5),
 ('featur', 5),
 ('green', 5),
 ('audienc', 5),
 ('children', 5),
 ('pink', 5),
 ('put', 5),
 ('femal', 5),
 ('nintendo', 4),
 ('red', 4),
 ('balloon', 4),
 ('happi', 4),
 ('moon', 4),
 ('stone', 4),
 ('includ', 4),
 ('plush', 4),
 ('toy', 4),
 ('design', 4),
 ('make', 4),
 ('like', 4),
 ('time', 4),
 ('introduc', 4),
 ('fall', 4),
 ('help', 4),
 ('two', 4)]

In [123]:
len(stemmed3)

1151

In [124]:
len(scount3)

580

In [125]:
len(set(stemmed3))

580

1151 tokens in a3.txt after removing stopwords and applying stemming

580 unique tokens in a3.txt after removing stopwords and applying stemming

In [126]:
stemmed4 = stem_tokens(filtered4, stemmer)
scount4 = Counter(stemmed4)
scount4.most_common(50)

[('charizard', 111),
 ('pokémon', 81),
 ('ash', 40),
 ('battl', 23),
 ('charact', 22),
 ('mega', 20),
 ('one', 20),
 ('appear', 19),
 ('red', 19),
 ('In', 18),
 ('game', 16),
 ('blue', 14),
 ('charmand', 14),
 ('team', 13),
 ('evolv', 12),
 ('charmeleon', 11),
 ('evolut', 11),
 ('X', 11),
 ('trainer', 11),
 ('origin', 9),
 ('form', 9),
 ('It', 9),
 ('power', 9),
 ('the', 9),
 ('anim', 8),
 ('Y', 8),
 ('seri', 8),
 ('state', 8),
 ('attack', 8),
 ('becom', 8),
 ('smash', 8),
 ('defeat', 8),
 ('orang', 7),
 ('popular', 7),
 ('pikachu', 7),
 ('leagu', 7),
 ('first', 6),
 ('they', 6),
 ('final', 6),
 ('featur', 6),
 ('describ', 6),
 ('super', 6),
 ('call', 6),
 ('children', 6),
 ('fire', 6),
 ('three', 6),
 ('video', 5),
 ('later', 5),
 ('adapt', 5),
 ('two', 5)]

In [127]:
len(stemmed4)

1792

In [128]:
len(scount4)

775

In [129]:
len(set(stemmed4))

775

1792 tokens in a4.txt after removing stopwords and applying stemming

775 unique tokens in a4.txt after removing stopwords and applying stemming

In [130]:
stemmed5 = stem_tokens(filtered5, stemmer)
scount5 = Counter(stemmed5)
scount5.most_common(50)

[('mewtwo', 121),
 ('pokémon', 59),
 ('charact', 30),
 ('In', 27),
 ('game', 23),
 ('the', 21),
 ('appear', 20),
 ('mew', 18),
 ('anim', 16),
 ('film', 16),
 ('voic', 15),
 ('clone', 14),
 ('japanes', 11),
 ('player', 11),
 ('use', 11),
 ('mega', 11),
 ('origin', 10),
 ('red', 9),
 ('also', 9),
 ('franchis', 8),
 ('creat', 8),
 ('first', 8),
 ('one', 8),
 ('seri', 8),
 ('children', 8),
 ('smash', 8),
 ('featur', 8),
 ('fuji', 8),
 ('nintendo', 7),
 ('later', 7),
 ('strike', 7),
 ('back', 7),
 ('other', 7),
 ('super', 7),
 ('bro', 7),
 ('includ', 7),
 ('media', 6),
 ('video', 6),
 ('blue', 6),
 ('learn', 6),
 ('releas', 6),
 ('power', 6),
 ('abil', 6),
 ('note', 6),
 ('defeat', 6),
 ('team', 6),
 ('rocket', 6),
 ('giovanni', 6),
 ('book', 6),
 ('adapt', 5)]

In [131]:
len(stemmed5)

1875

In [132]:
len(scount5)

836

In [133]:
len(set(stemmed5))

836

1875 tokens in a5.txt after removing stopwords and applying stemming

836 unique tokens in a5.txt after removing stopwords and applying stemming

In [134]:
stemmed6 = stem_tokens(filtered6, stemmer)
scount6 = Counter(stemmed6)
scount6.most_common(50)

[('meowth', 58),
 ('pokémon', 44),
 ('game', 17),
 ('appear', 17),
 ('anim', 14),
 ('charact', 12),
 ('In', 10),
 ('one', 9),
 ('first', 8),
 ('describ', 8),
 ('speci', 6),
 ('seri', 6),
 ('team', 6),
 ('speak', 6),
 ('the', 6),
 ('japanes', 5),
 ('languag', 5),
 ('nintendo', 5),
 ('episod', 5),
 ('rocket', 5),
 ('would', 5),
 ('coin', 5),
 ('state', 5),
 ('known', 4),
 ('franchis', 4),
 ('video', 4),
 ('red', 4),
 ('blue', 4),
 ('citi', 4),
 ('abil', 4),
 ('human', 4),
 ('attempt', 4),
 ('call', 4),
 ('meowzi', 4),
 ('two', 4),
 ('gamesradar', 4),
 ('use', 4),
 ('collect', 4),
 ('day', 4),
 ('captur', 4),
 ('editor', 4),
 ('version', 3),
 ('freak', 3),
 ('form', 3),
 ('It', 3),
 ('jame', 3),
 ('english', 3),
 ('show', 3),
 ('impress', 3),
 ('walk', 3)]

In [135]:
len(stemmed6)

941

In [136]:
len(scount6)

498

In [137]:
len(set(stemmed6))

498

941 tokens in a6.txt after removing stopwords and applying stemming

498 unique tokens in a6.txt after removing stopwords and applying stemming

In [138]:
stemmed7 = stem_tokens(filtered7, stemmer)
scount7 = Counter(stemmed7)
scount7.most_common(50)

[('gengar', 47),
 ('pokémon', 41),
 ('game', 15),
 ('appear', 12),
 ('In', 11),
 ('name', 9),
 ('red', 9),
 ('blue', 9),
 ('shadow', 8),
 ('best', 7),
 ('charact', 7),
 ('ghost', 7),
 ('the', 7),
 ('nintendo', 6),
 ('first', 6),
 ('call', 6),
 ('mega', 6),
 ('also', 6),
 ('haunter', 6),
 ('gym', 6),
 ('speci', 5),
 ('It', 5),
 ('design', 5),
 ('anim', 5),
 ('japanes', 5),
 ('one', 5),
 ('seri', 5),
 ('form', 5),
 ('citi', 5),
 ('use', 5),
 ('leader', 5),
 ('A', 5),
 ('ash', 5),
 ('later', 4),
 ('note', 4),
 ('children', 4),
 ('ign', 4),
 ('due', 4),
 ('evolut', 4),
 ('curs', 4),
 ('includ', 4),
 ('battl', 4),
 ('freak', 3),
 ('franchis', 3),
 ('sequel', 3),
 ('relat', 3),
 ('adapt', 3),
 ('popular', 3),
 ('prais', 3),
 ('magazin', 3)]

In [139]:
len(stemmed7)

886

In [140]:
len(scount7)

492

In [141]:
len(set(stemmed7))

492

886 tokens in a7.txt after removing stopwords and applying stemming

492 unique tokens in a7.txt after removing stopwords and applying stemming

In [142]:
stemmed8 = stem_tokens(filtered8, stemmer)
scount8 = Counter(stemmed8)
scount8.most_common(50)

[('jynx', 57),
 ('pokémon', 36),
 ('appear', 20),
 ('game', 19),
 ('design', 19),
 ('episod', 16),
 ('describ', 9),
 ('In', 9),
 ('black', 8),
 ('later', 7),
 ('anim', 7),
 ('critic', 7),
 ('purpl', 7),
 ('controversi', 7),
 ('red', 6),
 ('the', 6),
 ('charact', 6),
 ('also', 6),
 ('use', 6),
 ('lorelei', 6),
 ('list', 6),
 ('first', 5),
 ('japanes', 5),
 ('due', 5),
 ('one', 5),
 ('resembl', 5),
 ('seri', 5),
 ('ice', 5),
 ('air', 5),
 ('nintendo', 4),
 ('freak', 4),
 ('variou', 4),
 ('english', 4),
 ('blackfac', 4),
 ('chang', 4),
 ('origin', 4),
 ('color', 4),
 ('featur', 4),
 ('smoochum', 4),
 ('peopl', 4),
 ('note', 4),
 ('would', 4),
 ('remov', 4),
 ('rougela', 3),
 ('speci', 3),
 ('sugimori', 3),
 ('blue', 3),
 ('cultur', 3),
 ('weatherford', 3),
 ('gener', 3)]

In [143]:
len(stemmed8)

989

In [144]:
len(scount8)

528

In [145]:
len(set(stemmed8))

528

989 tokens in a8.txt after removing stopwords and applying stemming

528 unique tokens in a8.txt after removing stopwords and applying stemming

In [146]:
stemmed9 = stem_tokens(filtered9, stemmer)
scount9 = Counter(stemmed9)
scount9.most_common(50)

[('togepi', 60),
 ('pokémon', 37),
 ('appear', 17),
 ('In', 12),
 ('gold', 11),
 ('toget', 10),
 ('first', 9),
 ('togekiss', 9),
 ('anim', 8),
 ('egg', 8),
 ('game', 7),
 ('misti', 7),
 ('also', 7),
 ('evolv', 6),
 ('player', 6),
 ('describ', 6),
 ('It', 5),
 ('silver', 5),
 ('red', 5),
 ('blue', 5),
 ('pattern', 5),
 ('evolut', 5),
 ('back', 5),
 ('call', 5),
 ('one', 5),
 ('bodi', 4),
 ('shell', 4),
 ('retain', 4),
 ('give', 4),
 ('gain', 4),
 ('use', 4),
 ('design', 4),
 ('requir', 4),
 ('sinc', 4),
 ('attack', 4),
 ('sever', 4),
 ('featur', 4),
 ('comment', 4),
 ('fan', 4),
 ('charact', 3),
 ('video', 3),
 ('babi', 3),
 ('stubbi', 3),
 ('releas', 3),
 ('happi', 3),
 ('obtain', 3),
 ('wing', 3),
 ('diamond', 3),
 ('pearl', 3),
 ('second', 3)]

In [147]:
len(stemmed9)

867

In [148]:
len(scount9)

453

In [149]:
len(set(stemmed9))

453

867 tokens in a9.txt after removing stopwords and applying stemming

453 unique tokens in a9.txt after removing stopwords and applying stemming

In [150]:
stemmed10 = stem_tokens(filtered10, stemmer)
scount10 = Counter(stemmed10)
scount10.most_common(50)

[('snorlax', 45),
 ('pokémon', 31),
 ('In', 14),
 ('game', 11),
 ('call', 11),
 ('red', 9),
 ('use', 9),
 ('appear', 8),
 ('charact', 8),
 ('one', 8),
 ('later', 6),
 ('sleep', 6),
 ('ash', 6),
 ('also', 6),
 ('known', 5),
 ('anim', 5),
 ('ign', 5),
 ('power', 5),
 ('children', 5),
 ('first', 4),
 ('It', 4),
 ('adventur', 4),
 ('time', 4),
 ('the', 4),
 ('obtain', 4),
 ('player', 4),
 ('must', 4),
 ('defeat', 4),
 ('pokemon', 4),
 ('speci', 3),
 ('video', 3),
 ('blue', 3),
 ('variou', 3),
 ('voic', 3),
 ('seri', 3),
 ('pikachu', 3),
 ('main', 3),
 ('name', 3),
 ('relat', 3),
 ('bodi', 3),
 ('it', 3),
 ('larg', 3),
 ('play', 3),
 ('munchlax', 3),
 ('certain', 3),
 ('roadblock', 3),
 ('wake', 3),
 ('town', 3),
 ('poké', 3),
 ('slam', 3)]

In [151]:
len(stemmed10)

761

In [152]:
len(scount10)

427

In [153]:
len(set(stemmed10))

427

761 tokens in a10.txt after removing stopwords and applying stemming

427 unique tokens in a10.txt after removing stopwords and applying stemming

In [154]:
stemmedAll = stem_tokens(filteredAll, stemmer)
scountAll = Counter(stemmedAll)
scountAll.most_common(50)

[('pokémon', 521),
 ('ash', 189),
 ('appear', 163),
 ('game', 158),
 ('In', 150),
 ('charact', 144),
 ('mewtwo', 125),
 ('charizard', 114),
 ('anim', 111),
 ('the', 99),
 ('one', 97),
 ('red', 95),
 ('seri', 81),
 ('first', 73),
 ('also', 69),
 ('blue', 67),
 ('jigglypuff', 66),
 ('gyarado', 62),
 ('team', 61),
 ('togepi', 60),
 ('use', 59),
 ('meowth', 59),
 ('jynx', 57),
 ('battl', 55),
 ('evolv', 51),
 ('name', 50),
 ('describ', 49),
 ('design', 49),
 ('japanes', 47),
 ('episod', 47),
 ('gengar', 47),
 ('pikachu', 46),
 ('snorlax', 46),
 ('call', 45),
 ('It', 44),
 ('later', 43),
 ('mega', 43),
 ('player', 40),
 ('leagu', 40),
 ('origin', 39),
 ('nintendo', 38),
 ('video', 38),
 ('form', 37),
 ('featur', 37),
 ('defeat', 37),
 ('franchis', 36),
 ('voic', 35),
 ('children', 35),
 ('trainer', 33),
 ('power', 32)]

In [155]:
len(stemmedAll)

12826

In [156]:
len(scountAll)

2905

In [157]:
len(set(stemmedAll))

2905

12826 tokens in all 10 documents after removing stopwords and applying stemming

2905 unique tokens in all 10 documents after removing stopwords and applying stemming

In [158]:
import os

from sklearn.feature_extraction.text import TfidfVectorizer

path = 'Data'
token_dict = {}

def tokenize(text):
    tokens = nltk.word_tokenize(text)
    stems = stem_tokens(tokens, stemmer)
    return stems

for subdir, dirs, files in os.walk(path):
    for file in files:
        file_path = subdir + os.path.sep + file
        docs = open(file_path, 'r')
        text = docs.read()
        lowers = text.lower()
        no_punctuation = text.translate(str.maketrans('','',string.punctuation))
        token_dict[file] = no_punctuation

tfidf = TfidfVectorizer(tokenizer=tokenize)
tfs = tfidf.fit_transform(token_dict.values())

In [159]:
print(tfs)

  (0, 380)	0.0039054358464855115
  (0, 1492)	0.0039054358464855115
  (0, 890)	0.0039054358464855115
  (0, 2119)	0.0039054358464855115
  (0, 2898)	0.010563220155141247
  (0, 697)	0.010563220155141247
  (0, 566)	0.006984705485077477
  (0, 1334)	0.010563220155141247
  (0, 1733)	0.008979702187436498
  (0, 251)	0.008979702187436498
  (0, 1774)	0.015712356618584142
  (0, 1450)	0.008979702187436498
  (0, 2230)	0.015712356618584142
  (0, 2938)	0.015712356618584142
  (0, 2660)	0.010563220155141247
  (0, 1034)	0.007856178309292071
  (0, 1541)	0.0042776636392283015
  (0, 856)	0.006984705485077477
  (0, 2533)	0.008979702187436498
  (0, 1038)	0.0046891423738825745
  (0, 2260)	0.006272660341587323
  (0, 1437)	0.008979702187436498
  (0, 2677)	0.006984705485077477
  (0, 2692)	0.010563220155141247
  (0, 2690)	0.008979702187436498
  :	:
  (9, 2446)	0.004751235131980083
  (9, 1622)	0.008675597505774813
  (9, 2772)	0.004337798752887406
  (9, 1446)	0.008675597505774813
  (9, 2301)	0.004751235131980083
  (9

In [160]:
from sklearn.metrics.pairwise import cosine_similarity

In [168]:
print (cosine_similarity(tfs, tfs))

[[1.         0.52108035 0.57495528 0.52543025 0.55000814 0.50986548
  0.49197771 0.5166285  0.46469832 0.42765992]
 [0.52108035 1.         0.6304179  0.56802282 0.58311353 0.55305414
  0.52905401 0.55132745 0.49615439 0.4552163 ]
 [0.57495528 0.6304179  1.         0.63814194 0.73421915 0.66235883
  0.63194258 0.62527188 0.60545933 0.49244249]
 [0.52543025 0.56802282 0.63814194 1.         0.5952697  0.57175004
  0.53922256 0.56168587 0.5165055  0.47131129]
 [0.55000814 0.58311353 0.73421915 0.5952697  1.         0.60826746
  0.55731668 0.58191083 0.53697141 0.46087262]
 [0.50986548 0.55305414 0.66235883 0.57175004 0.60826746 1.
  0.54302076 0.55461249 0.51710996 0.44344059]
 [0.49197771 0.52905401 0.63194258 0.53922256 0.55731668 0.54302076
  1.         0.52466163 0.49354769 0.43304872]
 [0.5166285  0.55132745 0.62527188 0.56168587 0.58191083 0.55461249
  0.52466163 1.         0.50533253 0.44797345]
 [0.46469832 0.49615439 0.60545933 0.5165055  0.53697141 0.51710996
  0.49354769 0.50533

In [170]:
output_file = open('cosine_similarity.txt', 'w')
for w in (cosine_similarity(tfs, tfs)):
    print(w, file=output_file)